Buddies.API/Controllers/UsersController.cs

```diff
@@ -53,8 +63,76 @@ public ActionResult Register([FromBody] RegisterRequest request)
                 }
                 return ValidationProblem(ModelState);
             }
+
+            var user = _userManager.FindByEmailAsync(request.Email).Result;
+            var claims = _signInManager.CreateUserPrincipalAsync(user).Result;
+            
+            await HttpContext.SignInAsync(claims, new AuthenticationProperties
+            {
+                ExpiresUtc = DateTimeOffset.Now.AddDays(7),
+                IsPersistent = true
+            });
             
-            // todo: for login upon successful registration, send access token here (and it change to Created/201)
+            return StatusCode(StatusCodes.Status201Created);
+        }
+
+
+        /// <summary>
+        /// Endpoint for logging in a user.
+        /// </summary>
+        /// /// <param name="request">A login request.</param>
+        [HttpPost("[action]")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult> Login([FromBody] LoginRequest request)
+        {
+            var user =  _userManager.FindByEmailAsync(request.Email).Result;
+
+            if (user == null)
+            {
+                return Unauthorized("Wrong email or password"); 
+            }
+
+            var signInResult =  _signInManager.CheckPasswordSignInAsync(user, request.Password, false).Result;
+
+            if (!signInResult.Succeeded)
+            {
+                return Unauthorized("Wrong email or password!"); 
+            }
+
+            var claims = _signInManager.CreateUserPrincipalAsync(user).Result;
+
+            await HttpContext.SignInAsync(claims, new AuthenticationProperties
+            {
+                ExpiresUtc = DateTimeOffset.Now.AddDays(7),
+                IsPersistent = true // maybe add a "remember me" option in the future
+            });
+
+            return Ok();
+        }
+    
+        /// <summary>
+        /// Endpoint for renewing access tokens.
+        /// </summary>
+        [HttpGet("[action]")]
+        [Authorize(AuthenticationSchemes = CookieAuthenticationDefaults.AuthenticationScheme)]
+        public ActionResult<TokenResponse> Refresh()
+        {
+            var userEntity = _userManager.GetUserAsync(User).Result;
+            
+            var responseBody = new TokenResponse
+            {
+                AccessToken = _tokenService.GenerateAccessToken(userEntity)
+            };
+
+            return responseBody;
+        }
+        
+        [HttpGet("[action]")]
+        [Authorize]
+        public async Task<ActionResult> Logout()
```

Comment:
Docstring missing for logout method

Reply:
Andy mentioned during the previous meeting there is no need to document trivial portions of code.